sdk/samples/iot/provisioning/src/paho_iot_provisioning_example.c

```diff
@@ -50,161 +50,162 @@
-  {
-    return rc;
-  }
-
-  // Gracefully disconnect: send the disconnect packet and close the socket
-  if ((rc = MQTTClient_disconnect(mqtt_client, TIMEOUT_MQTT_DISCONNECT_MS)) != MQTTCLIENT_SUCCESS)
-  {
-    (void)printf("Failed to disconnect MQTT client: MQTTClient return code %d .\n", rc);
-    return rc;
-  }
-  (void)printf("Disconnected.\n");
-
-  // Clean up and release resources allocated by the mqtt client
-  MQTTClient_destroy(&mqtt_client);
-
-  return 0;
 }
 
-// Read the user environment variables used to connect to the Provisioning Service and initialize
-// the provisioning client
-static az_result read_configuration_and_init_client()
+static az_result read_environment_variables(
+    az_span* endpoint,
+    az_span* id_scope,
+    az_span* registration_id)
 {
-  az_span endpoint_span = AZ_SPAN_FROM_BUFFER(global_provisioning_endpoint);
+  // Certification variables
+  az_span cert = AZ_SPAN_FROM_BUFFER(x509_cert_pem_file_buffer);
+  AZ_RETURN_IF_FAILED(
+      read_configuration_entry(ENV_DEVICE_X509_CERT_PEM_FILE, NULL, false, cert, &cert));
+
+  az_span trust = AZ_SPAN_FROM_BUFFER(x509_trust_pem_file_buffer);
+  AZ_RETURN_IF_FAILED(
+      read_configuration_entry(ENV_DEVICE_X509_TRUST_PEM_FILE, "", false, trust, &trust));
+
+  // Connection variables
+  *endpoint = AZ_SPAN_FROM_BUFFER(global_provisioning_endpoint_buffer);
   AZ_RETURN_IF_FAILED(read_configuration_entry(
       ENV_GLOBAL_PROVISIONING_ENDPOINT,
       ENV_GLOBAL_PROVISIONING_ENDPOINT_DEFAULT,
       false,
-      endpoint_span,
-      &endpoint_span));
+      *endpoint,
+      endpoint));
 
-  az_span id_scope_span = AZ_SPAN_FROM_BUFFER(id_scope);
-  AZ_RETURN_IF_FAILED(
-      read_configuration_entry(ENV_ID_SCOPE_ENV, NULL, false, id_scope_span, &id_scope_span));
+  *id_scope = AZ_SPAN_FROM_BUFFER(id_scope_buffer);
+  AZ_RETURN_IF_FAILED(read_configuration_entry(ENV_ID_SCOPE_ENV, NULL, false, *id_scope, id_scope));
 
-  az_span registration_id_span = AZ_SPAN_FROM_BUFFER(registration_id);
+  *registration_id = AZ_SPAN_FROM_BUFFER(registration_id_buffer);
   AZ_RETURN_IF_FAILED(read_configuration_entry(
-      ENV_REGISTRATION_ID_ENV, NULL, false, registration_id_span, &registration_id_span));
-
-  az_span cert = AZ_SPAN_FROM_BUFFER(x509_cert_pem_file);
-  AZ_RETURN_IF_FAILED(
-      read_configuration_entry(ENV_DEVICE_X509_CERT_PEM_FILE, NULL, false, cert, &cert));
-
-  az_span trusted = AZ_SPAN_FROM_BUFFER(x509_trust_pem_file);
-  AZ_RETURN_IF_FAILED(
-      read_configuration_entry(ENV_DEVICE_X509_TRUST_PEM_FILE, "", false, trusted, &trusted));
-
-  // Initialize the provisioning client with the provisioning endpoint and the default connection
-  // options
-  AZ_RETURN_IF_FAILED(az_iot_provisioning_client_init(
-      &provisioning_client, endpoint_span, id_scope_span, registration_id_span, NULL));
+      ENV_REGISTRATION_ID_ENV, NULL, false, *registration_id, registration_id));
 
+  putchar('\n');
```

Comment:
what is this for?

Reply:
formatting in the print out of the sample.